File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-5.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        
        operators = {
            "+",
            "-",
            "*",
            "/"
        }
        stack = []
        for char in tokens:
            if char in operators:
                if char == "+":
                    b = stack.pop()
                    a = stack.pop()
                    stack.append(a + b)
                elif char == "-":
                    b = stack.pop()
                    a = stack.pop()
                    stack.append(a - b)
                elif char == "*":
                    b = stack.pop()
                    a = stack.pop()
                    stack.append(a * b)
                else:
                    b = stack.pop()
                    a = stack.pop()
                    stack.append(int(a / b))
            else:
                stack.append(int(char))
        
        return stack[-1]

File: Data Structures & Algorithms/validate-parentheses/submission-5.cpp
class Solution {
public:
    bool isValid(string s) {
        // std::set<char> openSet = {'(', '[', '{'}
        std::stack<char> stac;
        std::unordered_map<char, char> closedOpen = {
            {')', '('},
            {'}', '{'},
            {']', '['}
        };
        for (const auto& c: s) {
            if (closedOpen.contains(c)){
                if (!stac.empty()){
                    char openChar = stac.top();
                    if (!(closedOpen[c] == openChar)){
                        return false;
                    }
                    stac.pop();
                } else {return false;}
            } else {
                stac.push(c);
            }
        }
        return stac.empty();
    }
};
